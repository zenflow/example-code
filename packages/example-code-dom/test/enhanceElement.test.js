/* eslint-env jest */

const { enhanceElement } = require('../src/index')

const getCodeElement = () => {
  return Object.assign(document.createElement('code'), {
    className: 'executable js',
    innerHTML: 'window.callback()',
  })
}

describe('enhanceElement', () => {
  it('#01 snapshot', async () => {
    const codeElement = getCodeElement()

    expect(codeElement).toMatchSnapshot()
    enhanceElement(codeElement)
    expect(codeElement).toMatchSnapshot()
  })

  it('#02 throws when called on non-code elements', async () => {
    expect(() => enhanceElement(document.createElement('div'))).toThrowError()
  })

  it('#03 clicking button executes code', async () => {
    const codeElement = getCodeElement()
    enhanceElement(codeElement)

    let callbackCalled = false
    window.callback = () => {
      callbackCalled = true
    }

    codeElement.querySelector('button').click()

    expect(callbackCalled).toEqual(true)
  })

  it('#04 takes a executeButtonClass option', async () => {
    const codeElement = getCodeElement()
    enhanceElement(codeElement, { executeButtonClass: 'my-execute' })
    const button = codeElement.querySelector('button')
    expect(String(button.classList)).toEqual('my-execute')
  })
})
