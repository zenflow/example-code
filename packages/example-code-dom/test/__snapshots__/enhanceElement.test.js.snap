// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enhanceElement #01 snapshot 1`] = `
<code
  class="executable js"
>
  window.callback()
</code>
`;

exports[`enhanceElement #01 snapshot 2`] = `
<code
  class="executable js"
>
  <button
    class="execute"
    style="float: right;"
  >
    â–¶
  </button>
  window.callback()
</code>
`;
